sender/api/src/main/java/org/phenotips/integration/lims247/internal/RemoteSynchronizationEventListener.java

```diff
@@ -76,15 +93,86 @@
     @Inject
     private Execution execution;
 
+    /** Provides access to configuration file values. */
+    @Inject
+    @Named("xwikiproperties")
+    private ConfigurationSource configuration;
+
     /** HTTP client used for communicating with the remote server. */
     private final CloseableHttpClient client = HttpClients.createSystem();
 
+    /** Thread executor used for asynchronous requests. */
+    private ThreadPoolExecutor executor;
+
+    /** Holds HTTP request configuration to be used for different HTTP requests. */
+    private RequestConfig requestConfig;
+
+    /**
+     * Thread for individual asynchronous remote synchronization requests.
+     */
+    private class RemoteSynchronizationRequestThread extends AbstractXWikiRunnable {
+        /** Logging helper object. */
+        private Logger logger;
+
+        /** HTTP client used for communicating with the remote server. */
+        private final CloseableHttpClient client;
+
+        /** Request to send to the remote server. */
+        private final HttpPost request;
+
+        RemoteSynchronizationRequestThread(Logger logger, CloseableHttpClient client, HttpPost request) {
+            this.logger = logger;
+            this.client = client;
+            this.request = request;
+        }
+
+        @Override
+        public void runInternal() {
+            this.logger.debug("Sending remote sync request to [{}]", request.getURI());
+            try {
+                this.client.execute(this.request).close();
```

Comment:
Should ideally log the response or response status code

Reply:
As discussed, let's break out the task of generating (on the receiver side) and checking (on the sender side) the appropriate status code and message.